pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -114,6 +121,49 @@ func podKeyGet(pod *corev1.Pod) string {
 	return pod.Namespace + "/" + pod.Name
 }
 
+// buildVFDeviceIDListPerPod is a helper function to build a cache structure with the
+// list of all the PCI addresses allocated per Pod based on their resource requests (in Pod spec).
+// When there is a request for a VF resource (to associate it for a secondary network interface),
+// getUnusedSriovVFDeviceIDPerPod will use this cache information to pick up a unique PCI address
+// which is still not associated with a linux interface name.
+// NOTE: buildVFDeviceIDListPerPod is called only if a Pod specific VF to Interface mapping cache
+// was not build earlier. sample initial entry per pod: "{18:01.1,""},{18:01.2,""},{18:01.3,""}"
+func (pc *PodController) buildVFDeviceIDListPerPod(podName, podNameSpace string) error {
+	if pc.vfDeviceIDUsageMap[podName] != nil {
```

Comment:
We should use Namespace + podName as the key?

Reply:
![image](https://user-images.githubusercontent.com/70719963/190062596-97c3bea4-a615-42c5-997c-5d03384cd21a.png)

Reply:
could you clarify more on this please?.

Reply:
![image](https://user-images.githubusercontent.com/70719963/190062756-b133f1ad-8c0f-4da3-860f-5e5fe015c92d.png)

This is the visual representation of the cache.

Reply:
Just got to see your other comment as @jianjuns . use namespace + podname would be ideal too. I will modify it accordingly. Is that what this comment is about?.

Reply:
Sorry, my comments was not formatted correctly. I actually wrote:

`We should use <podNamespace>/<podName> as the key?`

Reply:
yes, this was taken care already and updated the PR in the morning @jianjuns